milli/src/update/settings.rs

```diff
@@ -481,6 +517,103 @@ mod tests {
         drop(rtxn);
     }
 
+    #[test]
+    fn default_stop_words() {
+        let path = tempfile::tempdir().unwrap();
+        let mut options = EnvOpenOptions::new();
+        options.map_size(10 * 1024 * 1024); // 10 MB
+        let index = Index::new(options, &path).unwrap();
+
+        // First we send 3 documents with ids from 1 to 3.
+        let mut wtxn = index.write_txn().unwrap();
+        let content = &b"name,age\nkevin,23\nkevina,21\nbenoit,34\n"[..];
+        let mut builder = IndexDocuments::new(&mut wtxn, &index, 0);
+        builder.update_format(UpdateFormat::Csv);
+        builder.execute(content, |_, _| ()).unwrap();
+        wtxn.commit().unwrap();
+
+        // Ensure there is no stop_words by default
+        let rtxn = index.read_txn().unwrap();
+        let stop_words = index.stop_words(&rtxn).unwrap();
+        assert!(stop_words.is_none());
+        drop(rtxn);
+    }
+
+    #[test]
+    fn set_and_reset_stop_words() {
+        let path = tempfile::tempdir().unwrap();
+        let mut options = EnvOpenOptions::new();
+        options.map_size(10 * 1024 * 1024); // 10 MB
+        let index = Index::new(options, &path).unwrap();
+
+        // First we send 3 documents with ids from 1 to 3.
+        let mut wtxn = index.write_txn().unwrap();
+        let content = &b"name,age,maxim\nkevin,23,I love dogs\nkevina,21,Doggos are the best\nbenoit,34,The crepes are really good\n"[..];
+        let mut builder = IndexDocuments::new(&mut wtxn, &index, 0);
+        builder.update_format(UpdateFormat::Csv);
+        builder.execute(content, |_, _| ()).unwrap();
+
+        // In the same transaction we provide some stop_words
+        let mut builder = Settings::new(&mut wtxn, &index, 0);
+        let mut set = std::collections::BTreeSet::new();
+        set.insert("i".to_string());
+        set.insert("the".to_string());
+        set.insert("are".to_string());
+        builder.set_stop_words(set.clone());
+        builder.execute(|_, _| ()).unwrap();
+        wtxn.commit().unwrap();
+
+        // Ensure there is no stop_words by default
+        let rtxn = index.read_txn().unwrap();
+        let stop_words = index.stop_words(&rtxn).unwrap();
+        assert!(stop_words.is_some()); // at this point the index should return something
+
+        let stop_words = stop_words.unwrap();
+        let expected = fst::Set::from_iter(&set).unwrap();
+        assert_eq!(stop_words.as_fst().as_bytes(), expected.as_fst().as_bytes());
+
+        // when we search for something that is a stop_words it should be ignored
+        let result = index.search(&rtxn).query("the").execute().unwrap();
+        assert!(result.documents_ids.is_empty());
+        let result = index.search(&rtxn).query("i").execute().unwrap();
+        assert!(result.documents_ids.is_empty());
+        let result = index.search(&rtxn).query("are").execute().unwrap();
+        assert!(result.documents_ids.is_empty());
```

Comment:
```suggestion
        // when we search for something that is a non prefix stop_words it should be ignored
        let result = index.search(&rtxn).query("the ").execute().unwrap();
        assert!(result.documents_ids.is_empty());
        let result = index.search(&rtxn).query("i ").execute().unwrap();
        assert!(result.documents_ids.is_empty());
        let result = index.search(&rtxn).query("are ").execute().unwrap();
        assert!(result.documents_ids.is_empty());
```

You are lucky here, you don't have any word that starts with one of your stop words.

Reply:
good catch, I understood that after writing this test :joy: